File: gencode_python/todo_2_claude-sonet-4.6.py
from http.server import BaseHTTPRequestHandler, HTTPServer
import json, uuid

todos: dict = {}

def make_todo(title: str) -> dict:
    tid = str(uuid.uuid4())
    todos[tid] = {"id": tid, "title": title, "completed": False}
    return todos[tid]

def send(handler, status: int, body):
    data = json.dumps(body).encode()
    handler.send_response(status)
    handler.send_header("Content-Type", "application/json")
    handler.send_header("Access-Control-Allow-Origin", "*")
    handler.send_header("Access-Control-Allow-Methods", "GET, POST, PUT, DELETE, OPTIONS")
    handler.send_header("Access-Control-Allow-Headers", "Content-Type")
    handler.end_headers()
    handler.wfile.write(data)

def read_body(handler) -> dict:
    length = int(handler.headers.get("Content-Length", 0))
    return json.loads(handler.rfile.read(length)) if length else {}

class TodoHandler(BaseHTTPRequestHandler):
    def log_message(self, *_): pass  # silence default logs

    def do_OPTIONS(self):
        send(self, 204, {})

    def do_GET(self):
        if self.path == "/todos":
            send(self, 200, list(todos.values()))
        elif self.path.startswith("/todos/"):
            tid = self.path.split("/")[-1]
            todo = todos.get(tid)
            send(self, 200, todo) if todo else send(self, 404, {"error": "Not found"})
        else:
            send(self, 404, {"error": "Not found"})

    def do_POST(self):
        if self.path == "/todos":
            body = read_body(self)
            title = (body.get("title") or "").strip()
            if not title:
                send(self, 400, {"error": "title is required"})
                return
            send(self, 201, make_todo(title))
        else:
            send(self, 404, {"error": "Not found"})

    def do_PUT(self):
        if self.path.startswith("/todos/"):
            tid = self.path.split("/")[-1]
            todo = todos.get(tid)
            if not todo:
                send(self, 404, {"error": "Not found"})
                return
            body = read_body(self)
            if "title" in body:
                todo["title"] = body["title"].strip() or todo["title"]
            if "completed" in body:
                todo["completed"] = bool(body["completed"])
            send(self, 200, todo)
        else:
            send(self, 404, {"error": "Not found"})

    def do_DELETE(self):
        if self.path.startswith("/todos/"):
            tid = self.path.split("/")[-1]
            todo = todos.pop(tid, None)
            send(self, 200, todo) if todo else send(self, 404, {"error": "Not found"})
        else:
            send(self, 404, {"error": "Not found"})

if __name__ == "__main__":
    server = HTTPServer(("localhost", 8000), TodoHandler)
    print("Todo API running at http://localhost:8000")
    print("  GET    /todos         — list all")
    print("  GET    /todos/:id     — get one")
    print("  POST   /todos         — create  { title }")
    print("  PUT    /todos/:id     — update  { title?, completed? }")
    print("  DELETE /todos/:id     — delete")
    server.serve_forever()

File: gencode_node/todo_2_claude-sonet-4.6.js
import http from "http";

let todos = [];
let nextId = 1;

function json(res, status, data) {
  res.writeHead(status, { "Content-Type": "application/json" });
  res.end(JSON.stringify(data));
}

function readBody(req) {
  return new Promise((resolve, reject) => {
    let body = "";
    req.on("data", (chunk) => (body += chunk));
    req.on("end", () => {
      try {
        resolve(body ? JSON.parse(body) : {});
      } catch {
        reject(new Error("Invalid JSON"));
      }
    });
  });
}

const server = http.createServer(async (req, res) => {
  const url = new URL(req.url, `http://${req.headers.host}`);
  const parts = url.pathname.replace(/^\/|\/$/g, "").split("/");
  const method = req.method;

  // GET /todos
  if (method === "GET" && parts[0] === "todos" && !parts[1]) {
    return json(res, 200, todos);
  }

  // GET /todos/:id
  if (method === "GET" && parts[0] === "todos" && parts[1]) {
    const todo = todos.find((t) => t.id === Number(parts[1]));
    return todo ? json(res, 200, todo) : json(res, 404, { error: "Not found" });
  }

  // POST /todos
  if (method === "POST" && parts[0] === "todos" && !parts[1]) {
    try {
      const body = await readBody(req);
      if (!body.title?.trim()) {
        return json(res, 400, { error: "title is required" });
      }
      const todo = { id: nextId++, title: body.title.trim(), completed: false };
      todos.push(todo);
      return json(res, 201, todo);
    } catch {
      return json(res, 400, { error: "Invalid JSON" });
    }
  }

  // PATCH /todos/:id
  if (method === "PATCH" && parts[0] === "todos" && parts[1]) {
    const idx = todos.findIndex((t) => t.id === Number(parts[1]));
    if (idx === -1) return json(res, 404, { error: "Not found" });
    try {
      const body = await readBody(req);
      if (body.title !== undefined) todos[idx].title = body.title.trim();
      if (body.completed !== undefined) todos[idx].completed = Boolean(body.completed);
      return json(res, 200, todos[idx]);
    } catch {
      return json(res, 400, { error: "Invalid JSON" });
    }
  }

  // DELETE /todos/:id
  if (method === "DELETE" && parts[0] === "todos" && parts[1]) {
    const idx = todos.findIndex((t) => t.id === Number(parts[1]));
    if (idx === -1) return json(res, 404, { error: "Not found" });
    todos.splice(idx, 1);
    return json(res, 204, null);
  }

  // DELETE /todos  (clear all)
  if (method === "DELETE" && parts[0] === "todos" && !parts[1]) {
    todos = [];
    return json(res, 204, null);
  }

  json(res, 404, { error: "Route not found" });
});

const PORT = process.env.PORT || 3000;
server.listen(PORT, () => console.log(`Todo server running on http://localhost:${PORT}`));
